en/Building_a_Simple_Engine/Loading_Models/08_animations.adoc

```diff
@@ -0,0 +1,1061 @@
+        lastTime = currentTime;
+
+        // Update model animations
+        animationTime += deltaTime;
+        if (!model.animations.empty()) {
+            model.updateAnimation(0, deltaTime);
+        }
+
+        drawFrame();
+    }
+
+    device.waitIdle();
+}
+----
+
+This code:
+
+1. Calculates the time elapsed since the last frame (deltaTime)
+2. Updates a global animation time counter (useful for custom animations)
+3. Calls `updateAnimation` on the model if it has animations
+4. Renders the frame with the updated animation state
+
+=== Advanced Animation Techniques
+
+While our basic animation system handles most common use cases, there are several advanced techniques you might want to implement:
+
+==== Animation Blending
+
+Animation blending is a technique that combines multiple animations to create smooth transitions or entirely new animations. This is essential for creating realistic character movement and responsive gameplay.
+
+===== Understanding Animation Blending
+
+At its core, animation blending works by interpolating between the transformations (position, rotation, scale) of corresponding bones or nodes in different animations. The key concepts include:
+
+* *Blend Factor*: A value between 0.0 and 1.0 that determines how much of each animation contributes to the final result
+* *Blend Space*: A multidimensional space where animations are positioned based on parameters (like speed, direction)
+* *Blend Trees*: Hierarchical structures that organize multiple blends into complex animation systems
+
+===== Types of Animation Blending
+
+There are several common types of animation blending:
+
+* *Linear Blending*: Simple interpolation between two animations (e.g., transitioning from walk to run)
+* *Additive Blending*: One animation is added on top of another (e.g., adding a "wounded" limp to any movement animation)
+* *Partial Blending*: Blending that affects only certain parts of the skeleton (e.g., aiming a weapon while walking)
+* *Parametric Blending*: Blending multiple animations based on continuous parameters (e.g., direction + speed)
+
+===== Implementing Basic Animation Blending
+
+Here's how to implement a simple linear blend between two animations:
+
+[source,cpp]
+----
+void blendAnimations(uint32_t fromIndex, uint32_t toIndex, float blendFactor) {
+    // Store original node transformations
+    std::vector<glm::vec3> originalTranslations;
+    std::vector<glm::quat> originalRotations;
+    std::vector<glm::vec3> originalScales;
+
+    for (auto node : model.linearNodes) {
+        originalTranslations.push_back(node->translation);
+        originalRotations.push_back(node->rotation);
+        originalScales.push_back(node->scale);
+    }
+
+    // Apply first animation fully
+    model.updateAnimation(fromIndex, 0.0f);
+
+    // Store intermediate transformations
+    std::vector<glm::vec3> fromTranslations;
+    std::vector<glm::quat> fromRotations;
+    std::vector<glm::vec3> fromScales;
```

Comment:
You should probably reserve `model.linearNodes.size()` elements for those three vectors.